### ISSUE TITLE
About inputs.tail

### ISSUE THREAD

**Issue by Ephemeraler** [influxdata/telegraf#14305]
### Please direct all support questsions to slack or the forums. Thank you.

I don't know why I can't see any metrics in output

I set inputs.tail
data_format = "grok"
  grok_patterns = ["%{CUSTOM_LOG}"]
  grok_custom_patterns = '''
  TYPE Failed password
  CUSTOM_LOG %{SYSLOGTIMESTAMP:timestamp} %{HOSTNAME:host:tag} %{PROG:prog:tag}\[%{NUMBER:number:drop}\]: %{TYPE:type} for %{WORD:user:tag} from %{IP:ip:tag} port %{NUMBER:port:tag}'''

and I run telegraf in debug.
2023-11-16T07:27:44Z D! [inputs.tail] Tail added for "/root/zhaofk/muqali/testlog"
2023-11-16T07:27:49Z D! [outputs.file] Wrote batch of 4 metrics in 75.369µs
2023-11-16T07:27:49Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-11-16T07:27:54Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics

I can see 4 metrics written into batch, but why I can't see any metrics

**Comment by Ephemeraler** [comment]
Oh, I find when I set outputs.file data_format = "influx", it works. but when I set prometheus, it can't print metric.

**Comment by Ephemeraler** [comment]
When I change %{NUMBER:port:tag} to %{NUMBER:value:int} in CUSTOM_LOG, outputs.prometheus could work. sry, I don't find relevant specification.

**Comment by Ephemeraler** [comment]
In order to use outputs.prometheus, it seems like I must set a modifier to int. is it right? If there isn't value(int) in my log, what should I do?

**Comment by Hipska** [comment]
I think Prometheus simply only supports numerical for their metrics.. You could add a fixed number maybe?

**Comment by powersj** [comment]
Yes, Prometheus metrics require a numeric value and cannot be all strings.

Closing this as this is not a forum for support.
